.github/workflows/e2e-test.yml

```diff
@@ -0,0 +1,26 @@
+name: E2E Test in Docker
+
+on:
+  pull_request:
+    types: [labeled, opened, synchronize, reopened]
+
+jobs:
+  e2e:
+    name: Run Makefile in e2e container
+    runs-on: [self-hosted]
+
+    if: contains(github.event.pull_request.labels.*.name, 'ok-to-test')
```

Comment:
Oh, I see. You want to run CI when PR has a label. Then you'll want to trigger it when it's labeled, too. It's a complicated thing in GitHub CI. :)